base/docs/basedocs.jl

```diff
@@ -130,81 +123,97 @@ declared to be a subtype of some supertype.
 kw"primitive type"
 
 """
+    macro
+
 `macro` defines a method to include generated code in the final body of a program. A
 macro maps a tuple of arguments to a returned expression, and the resulting expression
 is compiled directly rather than requiring a runtime `eval` call. Macro arguments may
 include expressions, literal values, and symbols. For example:
 
-    macro sayhello(name)
-        return :( println("Hello, ", \$name) )
-    end
+# Examples
+```jldoctest
+julia> macro sayhello(name)
+           return :( println("Hello, ", \$name, "!") )
+       end
+@sayhello (macro with 1 method)
 
-This macro takes one argument: `name`. When `@sayhello` is encountered, the quoted
-expression is expanded to interpolate the value of the argument into the final
-expression.
+julia> @sayhello "Charlie"
+Hello, Charlie!
+```
 """
 kw"macro"
 
 """
-`importall` imports all names exported by the specified module, as if `import` were used
-individually on all of them.  For example:
-
-    importall Distributions
+    importall
 
-As with `import`, functions imported by `importall` can be extended.
+`importall` imports all names exported by the specified module,
+as if `import` were used individually on all of them.
+See the [manual section about modules](@ref modules) for details.
 """
 kw"importall"
 
 """
-`local` introduces a new local variable. For example:
+    local
 
-    function foo(n)
-        x = 0
-        for i = 1:n
-            local x
-            x = i
-        end
-        x
-    end
+`local` introduces a new local variable.
 
-    julia> foo(10)
-    0
-
-Here `local x` introduces a separate `x` inside the loop, so the function returns `0`.
+# Examples
+```jldoctest
+julia> function foo(n)
+           x = 0
+           for i = 1:n
+               local x # introduce a loop-local x
+               x = i
+           end
+           x
+       end
+foo (generic function with 1 method)
+
+julia> foo(10)
+0
```

Comment:
This is a really good example, since it displays not only the use of `local` but also an important effect it has. Nice job!